code/game/objects/structures/signs/_signs.dm

```diff
@@ -8,6 +10,25 @@
 	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50, "stamina" = 0)
 	var/buildable_sign = 1 //unwrenchable and modifiable
 	rad_flags = RAD_PROTECT_CONTENTS | RAD_NO_CONTAMINATE
+	var/random_dept_base  // will be replaced by a random subtype if the station has random_dept and this is defined
```

Comment:
I don't get the usage of this var.

Reply:
iit defines what subtypes are picked from, so the arrow signs (/obj/structure/sign/directional) dont pick from a full sign (/obj/structure/sign/departments)

Reply:
Remove this var, and you need to declare two static list to contain two subtypes. One will be dept sign, the another one will be direction sign. Randomize() should pick one through if type check from the list.
This will be like:
```
var/static/list/sign_dept = subtypesof(dept ones)
var/static/list/sign_dir = subtypesof(direction ones)
var/obj/structure/sign/picked = istype(src, /obj/structure/sign/departments) ? pick(sign_dept) : pick(sign_dir)
```